sdk/communication/Azure.Communication.Administration/src/Models/PhoneNumberReservationOperation.cs

```diff
@@ -0,0 +1,176 @@
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        public override Response UpdateStatus(CancellationToken cancellationToken = default)
+            => UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+            => await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        /// <summary>
+        /// Check for the latest status of the operation.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <param name="async" />
+        /// <returns>The <see cref="Response"/> with the status update.</returns>
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken)
+        {
+            using DiagnosticScope scope = _client.ClientDiagnostics.CreateScope($"{nameof(PhoneNumberReservationOperation)}.{nameof(UpdateStatus)}");
+            scope.Start();
+
+            try
+            {
+                if (HasCompleted)
+                {
+                    return GetRawResponse();
+                }
+
+                if (cancellationToken == default)
+                {
+                    cancellationToken = _cancellationToken;
+                }
+
+                Response<PhoneNumberReservation> update = async
+                    ? await _client.GetReservationByIdAsync(reservationId: Id, cancellationToken: cancellationToken).ConfigureAwait(false)
+                    : _client.GetReservationById(reservationId: Id, cancellationToken: cancellationToken);
+
+                var response = update.GetRawResponse();
+                _rawResponse = response;
+
+                if (IsOperationCompleted(update))
+                {
+                    lock (_lockObject)
+                    {
+                        _rawResponse = response;
+                        _value = update.Value;
+                        _hasCompleted = true;
+                    }
+                }
+
+                return _rawResponse;
```

Comment:
Let's consistently use `GetRawResponse()`
(after we've got rid of the `_finalRawResponse` if else inside it)